pkg/resource/v1/chartconfig/desired.go

```diff
@@ -2,8 +2,70 @@ package chartconfig
 
 import (
 	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/microerror"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/cluster-operator/pkg/cluster"
+	"github.com/giantswarm/cluster-operator/pkg/label"
+)
+
+const (
+	chartConfigAPIVersion           = "core.giantswarm.io"
+	chartConfigKind                 = "ChartConfig"
+	chartConfigVersionBundleVersion = "0.1.0"
+
+	chartOperatorChart           = "quay.io/giantswarm/chart-operator-chart"
+	chartOperatorChartConfigName = "chart-operator"
+	chartOperatorChannel         = "stable"
+	chartOperatorRelease         = "chart-operator"
 )
 
+// GetDesiredState returns all desired ChartConfigs for managed guest resources.
 func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	clusterGuestConfig, err := r.toClusterGuestConfigFunc(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	clusterConfig, err := prepareClusterConfig(*r.baseClusterConfig, *clusterGuestConfig)
```

Comment:
Is there a reason why we have pointers if we are not keen to use them here? We may want to just have simple values by default for these kind if things.

Reply:
I think this is from current `certconfig` resource. My thinking here is that when creating external interface, I'd use pointer for structs unless something really requires actual value use. This because use cases for external interface are not always known and this has performance implications. Within resource implementation, in this specific case `prepareClusterConfig` requires actual values for correctness reasons: If pointers were used as parameters, there would be possibility for mutation of original value which is not wanted.

Reply:
Simple function calls with values should be optimized by compiler anyway. Also performance penalty is negligible if there isn't any loop creating > 1M copies. And even in that case passing values is often cheaper anyway as GC doesn't have to track dependencies. So performance gain is questionable IMO.

I'm with @xh3b4sd here. I believe passing "pure data" structures as values is good, as we gain semi-immutability for free. Also the intention is clear for readers.

Reply:
Ok. Let's refactor this in a moment once we get on going PRs merged.

Reply:
Thanks, I'll do the refactoring here.

Reply:
I've updated to pass values rather than pointers.